### PR TITLE
Support IntelliJ 2023.3

### PR DESCRIPTION
**Pull request by peddiashrith**
### Notes
1. adds GitHub actions changes
2. add plugin info in the `build.gradle.kts`
3. adds `IC-233` source folder (copied from `IC-232` folder)

Changes referenced from: https://github.com/amazon-ion/ion-intellij-plugin/pull/56

### Testing
- `./gradlew buildPlugin`
- `./gradlew verifyPlugin`
- Tested that generated zip file is working with IntelliJ `2023.3.1` version

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -18,7 +18,7 @@ jobs:
   release:
     strategy:
       matrix:
-        product: [ "IC-2022.2", "IC-2023.1", "IC-2023.2" ]
+        product: [ "IC-2022.2", "IC-2023.1", "IC-2023.2", "IC-2023.3"]
```
Comment (desaikd) [inline .github/workflows/release.yml:21]:
```suggestion
        product: [ "IC-2022.2", "IC-2023.1", "IC-2023.2", "IC-2023.3" ]
```

File: .github/workflows/pr.yml [hunk 2]
```diff
@@ -34,7 +34,7 @@ jobs:
     runs-on: ubuntu-latest
     strategy:
       matrix:
-        product: [ "IC-2022.2", "IC-2023.1", "IC-2023.2"]
+        product: [ "IC-2022.2", "IC-2023.1", "IC-2023.2", "IC-2023.3"]
```
Comment (desaikd) [inline .github/workflows/pr.yml:37]:
```suggestion
        product: [ "IC-2022.2", "IC-2023.1", "IC-2023.2", "IC-2023.3" ]
```

**Review by desaikd — APPROVED** [review]
Thank you for your contribution @peddiashrith!
Some minor comments to address, otherwise the changes looks good. 🚢 
I have verified locally with `./gradlew runIde` as well to check formatting of an Ion file content.

**Comment by peddiashrith** [comment]
Update the pull request with requested changes

**Pull request merged**